lib/segment_snippet.rb

```diff
@@ -0,0 +1,74 @@
+class SegmentSnippet
+  def initialize(segment_write_key = ENV['SEGMENT_WRITE_KEY'])
+    @segment_write_key = segment_write_key
+  end
+
+  def setup_segment_head
+    return if segment_write_key.blank?
+
+    <<-HTML
+      (function () {
```

Comment:
Have you written it yourself or given in some Segment documentation?

Reply:
Nope, got it from Segment itself :sweat_smile:. It's just their JS lib really, but pupilfirst implements these JS snippets as part of the ruby code for some reason. Probably a legacy aspect from previous versions of rails.